src/null-safety/faq.md

```diff
@@ -319,72 +319,75 @@ and situations where a null value is really expected. So the tool tells you what
+should not affect the compiler output much.
 
 A few notes that are worth highlighting:
 
-* dart2js emits `!` null assertions, but you may not notice them.
-  That's because pre null-safety dart2js already emitted null checks
-  (so they are already in the existing programs).
-
-* dart2js emits these null assertions regardless of sound/unsound null safety
-  and regardless of optimization level. In fact, dart2js doesn't remove
-  `!` when using `-O3` or `--omit-implicit-checks`.
-
-* dart2js may optimize away unnecessary null checks. This is because the same
-  optimizations done by dart2js in the past will be able to eliminate
-  unnecessary null checks when it knows the value is not null.
-
-* By default dart2js would generate parameter subtype checks (runtime checks
-  used to ensure covariant virtual calls are given appropriate arguments).
-  These are elided with the `--omit-implicit-checks` flag just as before.
-  Recall that this flag can make programs have unexpected behavior if types
-  are invalid, so we continue to recommend that the code has strong test
-  coverage to avoid any surprises. In particular, dart2js optimizes code based
+* The compiler generates `!` null assertions, but you might not notice
+  them. Pre null-safety, the compiler generated null checks that still
+  remain in existing programs).
+
+* The compiler generates these null assertions regardless of the
+  soundness of null safety or optimization level. In fact, the compiler
+  doesn't remove `!` when using `-O3` or `--omit-implicit-checks`.
+
+* The compiler might remove unnecessary null checks. Previous compiler
+  optimizations can remove unnecessary null checks when it knows the
+  value is not null.
+
+* By default, the compiler would generate parameter subtype checks.
+  These runtime checks ensure covariant virtual calls have appropriate
+  arguments. The compiler skips these checks with the
+  `--omit-implicit-checks` option. Using this option can generate apps
+  with unexpected behavior if the code includes invalid types.
+  To avoid any surprises, continue provide strong test coverage for
+  your code. In particular, the compiler optimizes code based
   on the fact that inputs should comply with the type declaration. If
-  the code provides arguments of an invalid type, those optimizations would
-  be wrong and the program could misbehave. This was true for inconsistent
-  types before, and is true with inconsistent nullabilities now with sound
-  null-safety.
+  the code provides arguments of an invalid type, those optimizations
+  would be wrong and the program could misbehave. This was true for
+  inconsistent types before, and is true with inconsistent 
+  nullabilities now with sound null-safety.
 
-* You may notice that DDC and the Dart VM have special error
-  messages for null checks, but to keep applications small dart2js does not.
+* You may notice that the development mode of the compiler and the Dart 
```

Comment:
nit: replace "the development mode of the compiler" with "the web development compiler".